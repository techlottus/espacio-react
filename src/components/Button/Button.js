

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


