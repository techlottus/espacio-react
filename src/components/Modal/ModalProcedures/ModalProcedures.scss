@import '../../../App.scss';
@import "../Modal.scss";

.modal-academic {
  @extend .modal-container;
  .modalbody {
    
    .academicprofile {
      margin: pixeltorem(16px) 0;
    }
    .infoacademic {
      font-family: "Nunito", sans-serif;
      font-size: pixeltorem(14px);
      line-height: pixeltorem(19px);
      padding-bottom: pixeltorem(16px);
      .infoacademictext {
        & .material-icons {
          padding-right: pixeltorem(11px);
        }
        &:last-child {
          padding-bottom: pixeltorem(16px);
        }
      }
    }
    .modalbtns {
      @media only screen and (min-width: map-get($grid-breakpoints, s)) {
        float: right;
      }
    }

    .contentfeedbackprocedures {
      font-family: "Nunito Bold", sans-serif;
      font-size: pixeltorem(18px);
      line-height: pixeltorem(24px);
    }
  }
}
