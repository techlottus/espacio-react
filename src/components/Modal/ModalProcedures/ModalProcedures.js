import React, { useEffect, useRef, useState } from "react";
import { Button } from "../../Button/Button";
import {
  btnModalAcademicContinue,
  btnModalAcademicExit,
} from "../../../constants/ProceduresAcademicHistory.constant";
import "./ModalProcedures.scss";
import { cardProfile } from "../../../constants/AccountStatus.constant";
import { Feedback } from "../../Feedback/Feedback";
import { CardProfile } from "../../Card/CardProfile";
import { feedbackProcedures } from "../../../constants/Procedures.constant";
import { useSelector } from "react-redux";
import { getImageOfAssets } from "../../../helpers/getImages";

export const ModalProcedures = ({
  isShow,
  onClose,
  onNext,
  infoModal,
  outside,
}) => {
  const [isModal, setIsModal] = useState(false);
  const { texts } = useSelector((state) => state.texts);

  const modalRef = useRef();

  const [profile, setProfile] = useState({
    ...cardProfile,
    image: getImageOfAssets(texts?.accountStatus?.images?.user),
  });

  const { profile: profileProcedure } = useSelector(
    (state) => state.procedures
  );

  useEffect(() => {
    if (infoModal.profile !== null) {
      const inforProfile = infoModal.profile;
      setProfile((state) => {
        return {
          ...state,
          name: inforProfile?.name,
          items: inforProfile?.items,
        };
      });
    }
  }, [infoModal.profile]); // eslint-disable-line react-hooks/exhaustive-deps

  const handleFeedback = () => {
    return (
      <div className="contentfeedbackprocedures">
        {infoModal.title} {texts?.procedures?.modalProcedures?.feedback}{" "}
        {infoModal.cost}
      </div>
    );
  };
  useEffect(() => {
    setIsModal(isShow);
  }, [isShow]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    if (isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [isShow]);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target) &&
        window.innerWidth > 991
      ) {
        outside();
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const handleModal = () => {
    return (
      <div className="modal-academic">
        <div className="modalcontent" ref={modalRef}>
          <div className="modalheader">
            <div className="title">
              {texts?.procedures?.modalProcedures?.title}
            </div>
          </div>
          <div className="modalbody">
            {profileProcedure ? (
              <div className="academicprofile">
                <CardProfile data={profile} />
              </div>
            ) : (
              ""
            )}
            <div className="infoacademic">
              {infoModal.information.map((info, i) => {
                return (
                  <p key={i} className="infoacademictext">
                    {info}
                  </p>
                );
              })}
              {infoModal?.file && Array.isArray(infoModal.file)
                ? infoModal.file.map((doc, i) => {
                    return (
                      <p key={i} className="infoacademictext">
                        <span className="material-icons outlined">
                          description
                        </span>
                        {doc}
                      </p>
                    );
                  })
                : ""}
            </div>
            {infoModal.title ? (
              <div className="feedbackcontent">
                <Feedback data={feedbackProcedures} html={handleFeedback} />
              </div>
            ) : (
              <></>
            )}

            <div className="modalbtns">
              <div className="btnsecundary">
                <Button
                  data={{
                    ...btnModalAcademicExit,
                    isExpand: window.innerWidth < 991,
                  }}
                  onClick={() => {
                    onClose();
                  }}
                />
              </div>
              <div className="btnprimary">
                <Button
                  data={{
                    ...btnModalAcademicContinue,
                    isExpand: window.innerWidth < 991,
                  }}
                  onClick={() => {
                    onNext();
                  }}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  };

  return <>{isModal ? handleModal() : ""}</>;
};
