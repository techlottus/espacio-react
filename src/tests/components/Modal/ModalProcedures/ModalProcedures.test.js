import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { ModalProcedures } from "../../../../components/Modal/ModalProcedures/ModalProcedures";
import { act } from "react-dom/test-utils";
import { Button } from "../../../../components/Button/Button";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState = {
    texts: {},
    procedures: {
      profile: {
        name: 'Jorge Hernandez',
        items: [
          '200029578',
          'Utc distancia',
          'Licenciatura 334',
          'Administración de empresas'
        ]
      },
    }
}

const isShow = true;
const onClose = jest.fn();
const onNext = jest.fn();
const infoModal = {"profile":{"name":"Jorge Hernandez","items":["200029578","Utc distancia","Licenciatura 334","Administración de empresas"]},"information":["Contacto: 5530091000","Constancia: $480.00 - Constancia de Término 100%","Tipo de entrega: Únicamente en Formato Digital"],"file":[],"title":"Constancia de estudio","cost":"$480.00","detailId":"1204"};
const outside = jest.fn();

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <ModalProcedures isShow={isShow} onClose={onClose} onNext={onNext} infoModal={infoModal} outside={outside}/>
    </Provider>
  );
};

describe("Test <ModalProcedures/>", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test("should close modal", () => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(Button).first().props().onClick();
    });
    wrapper.update();
  });

  test("should click next modal", () => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(Button).last().props().onClick();
    });
    wrapper.update();
  });

//   test("should close on outside modal", () => {
//     const wrapper = handleMount(initState);
//     wrapper.update();
//   });

});
