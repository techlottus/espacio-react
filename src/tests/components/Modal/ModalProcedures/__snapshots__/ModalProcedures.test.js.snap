// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ModalProcedures/> should init component 1`] = `
<div
  className="modal-academic"
>
  <div
    className="modalcontent"
  >
    <div
      className="modalheader"
    >
      <div
        className="title"
      />
    </div>
    <div
      className="modalbody"
    >
      <div
        className="academicprofile"
      >
        <lottus-card-profile />
      </div>
      <div
        className="infoacademic"
      >
        <p
          className="infoacademictext"
          key="0"
        >
          Contacto: 5530091000
        </p>
        <p
          className="infoacademictext"
          key="1"
        >
          Constancia: $480.00 - Constancia de Término 100%
        </p>
        <p
          className="infoacademictext"
          key="2"
        >
          Tipo de entrega: Únicamente en Formato Digital
        </p>
      </div>
      <div
        className="feedbackcontent"
      >
        <lottus-feedback>
          <div
            className="feedback-content "
            content=""
          >
            <div
              className="contentfeedbackprocedures"
            >
              Constancia de estudio
               
               
              $480.00
            </div>
          </div>
        </lottus-feedback>
      </div>
      <div
        className="modalbtns"
      >
        <div
          className="btnsecundary"
        >
          <lottus-button />
        </div>
        <div
          className="btnprimary"
        >
          <lottus-button />
        </div>
      </div>
    </div>
  </div>
</div>
`;
